game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/Retreater.java

```diff
@@ -0,0 +1,32 @@
+package games.strategy.triplea.delegate.battle.steps.retreat;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import java.util.Collection;
+import java.util.Map;
+
+interface Retreater {
```

Comment:
I think a javadoc would do well here to explain what this abstraction is representing exactly. WDYT?

Reply:
I mentioned when I first created this PR about turning these into "change objects".  I don't think it fits with any of the existing "change" interfaces (`Change`, `HistoryNode` are what I've found) because it either can't invert (such as `Change`) or it isn't just a history text (such as `HistoryNode`).  Maybe at some point, we might want to allow it to invert so being an instance of `Change` would be best.  WDYT?

Reply:
If invert becomes a big problem, then we can probably compensate that by replaying the n-1 events. I think eventually we'll have a base-state that will be fixed that would allow us to do that. Overall I think the history probably should not be displaying every single change and changes would probably become more granular.

I think there is some groundwork that needs to be done before we can get there, so I wouldn't force objects to be 'change' objects just yet unless it really fits.